hail/src/main/scala/is/hail/annotations/Region.scala

```diff
@@ -193,15 +203,15 @@ final class Region private (empty: Boolean) extends NativeBase() {
     nativeSetParentReference(r, i)
   }
 
-  def setFromDependentRegion(base: Region, i: Int): Unit = {
+  def setFromDependentRegion(base: Region, i: Int, blockSize: Int): Unit = {
     assert(i < nativeGetNumParents())
-    base.nativeGetParentReferenceInto(this, i)
+    base.nativeGetParentReferenceInto(this, i, blockSize)
   }
 
-  def getParentReference(i: Int): Region = {
+  def getParentReference(i: Int, blockSize: Int): Region = {
     assert(i < nativeGetNumParents())
-    val r = new Region(empty = true)
-    nativeGetParentReferenceInto(r, i)
+    val r = new Region(blockSize)
+    nativeGetParentReferenceInto(r, i, blockSize)
```

Comment:
I'm following up with a rewrite of aggregator stuff that will rewrite how this works, at which point this weirdness should disappear.

It turns out that the bulk of the time spent in the region constructor is actually just in creating the native Region object, so the mess caused by allowing an invalid region didn't seem worth it since creating the bare region object costs about the same as creating an actual region.

Reply:
Okay, thanks for explaining!